### PR TITLE
fix: casting to ARRAY types failed

### LINKED ISSUE #9440: Casting to ARRAY types failed
### Describe the bug

When performing a SQL `CAST` operation and the target is an ARRAY type.

### To Reproduce

Run `select [1,2,3]::int[]` in CLI.

```sh
DataFusion CLI v36.0.0

❯ select [1,2,3]::int[];
Optimizer rule 'simplify_expressions' failed
caused by
simplify_expressions
caused by
Internal error: Failed due to a difference in schemas, original schema: DFSchema { fields: [DFField { qualifier: None, field: Field { name: "make_array(Int64(1),Int64(2),Int64(3))", data_type: List(Field { name: "item", data_type: Int32, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }), nullable: false, dict_id: 0, dict_is_ordered: false, metadata: {} } }], metadata: {}, functional_dependencies: FunctionalDependencies { deps: [] } }, new schema: DFSchema { fields: [DFField { qualifier: None, field: Field { name: "make_array(Int64(1),Int64(2),Int64(3))", data_type: List(Field { name: "field", data_type: Int32, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }), nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} } }], metadata: {}, functional_dependencies: FunctionalDependencies { deps: [] } }.
This was likely caused by a bug in DataFusion's code and we would welcome that you file an bug report in our issue tracker

❯ select cast([1,2,3] as int[]);
Optimizer rule 'simplify_expressions' failed
caused by
simplify_expressions
caused by
Internal error: Failed due to a difference in schemas, original schema: DFSchema { fields: [DFField { qualifier: None, field: Field { name: "make_array(Int64(1),Int64(2),Int64(3))", data_type: List(Field { name: "item", data_type: Int32, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }), nullable: false, dict_id: 0, dict_is_ordered: false, metadata: {} } }], metadata: {}, functional_dependencies: FunctionalDependencies { deps: [] } }, new schema: DFSchema { fields: [DFField { qualifier: None, field: Field { name: "make_array(Int64(1),Int64(2),Int64(3))", data_type: List(Field { name: "field", data_type: Int32, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }), nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} } }], metadata: {}, functional_dependencies: FunctionalDependencies { deps: [] } }.
This was likely caused by a bug in DataFusion's code and we would welcome that you file an bug report in our issue tracker
```

### Expected behavior

These casts should be successful.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jonahgao**
## Which issue does this PR close?

Closes #9440.

## Rationale for this change

`int[]`  has been converted to `DataType::List`. However, the inner field should be renamed to 'item' instead of 'field' to ensure better integration with the existing ARRAY implementation. For instance, the `arrow_cast` function uses 'item'.
https://github.com/apache/arrow-datafusion/blob/c43d9b347ab17f458150ffbc1339b4e63d9e8b43/datafusion/sql/src/expr/arrow_cast.rs#L166-L167

## What changes are included in this PR?
Change the name of the inner field for `List` type within the SQL planner.

## Are these changes tested?
Yes

## Are there any user-facing changes?

No

### CODE REVIEW

File: datafusion/sqllogictest/test_files/array.slt [hunk 1]
```diff
@@ -6213,12 +6213,19 @@ select * from test_create_array_table;
 query T
 select arrow_typeof(a) from test_create_array_table;
 ----
-List(Field { name: "field", data_type: Int32, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} })
+List(Field { name: "item", data_type: Int32, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} })
```
Comment (jonahgao) [inline datafusion/sqllogictest/test_files/array.slt:6216]: It also fixed the list type defined through `create table`.

File: datafusion/sql/src/planner.rs [hunk 2]
```diff
@@ -362,7 +362,7 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
                 let data_type = self.convert_data_type(inner_sql_type)?;
 
                 Ok(DataType::List(Arc::new(Field::new(
-                    "field", data_type, true,
+                    "item", data_type, true,
```
Comment (alamb) [inline datafusion/sql/src/planner.rs:365]: Maybe we could use `DataType::new_list_field()` https://docs.rs/arrow/latest/arrow/datatypes/struct.Field.html#method.new_list_field

**Review by alamb — APPROVED** [review]
Thank you @jonahgao  🙏

File: datafusion/sql/src/planner.rs (hunk 2 above)
Comment (jonahgao) [reply datafusion/sql/src/planner.rs]: Updated. Thank you @alamb , I didn't notice this convenient API.

**Comment by alamb** [comment]
Thanks @jonahgao

**Pull request merged**